### PR TITLE
Fix redirect loop

### PR DESCRIPTION
**Pull request by jujinfu**
When a user  trying to visit a form without permission, this will show
an error msg and put user back to it's profile page.

### CODE REVIEW

File: user.php [hunk 1]
```diff
@@ -217,4 +217,27 @@
...
+	 *Added By Jinfu Feb 2015
+	 */
+	case 'switchuser':
+		if(!icms::$user){
+			redirect_header('index.php',5,_US_NOPERMISS);
+		} elseif (!empty($_GET['xoops_redirect'])) {
+			$redirect = htmlspecialchars(trim($_GET['xoops_redirect']));
+			$isExternal = FALSE;
+			$tragetURL=substr(ICMS_URL,0,strpos(ICMS_URL,"/formulize")).$redirect;
+			redirect_header("user.php?op=main",5, "Visiting: '".$tragetURL."'<br />"._US_NOPERMISS."<br />You may switch user and try again.");
```
Comment (eikonos) [inline user.php]: In most cases, users only have one account, so if they can't view a form, they can't switch users. The message should say something like "You do not have permission to view this page."

Comment (jujinfu) [inline user.php]: should i redirect to user.php?op=main or userinfor.php?uid=.........

**Comment by jujinfu** [comment]
:+1:

File: user.php [hunk 2]
```diff
@@ -217,4 +217,29 @@
...
+	 *Added By Jinfu Feb 2015
+	 */
+	case 'nopermission':
+		if(!icms::$user){
+			redirect_header('index.php',5,_US_NOPERMISS);
+		} elseif (!empty($_GET['xoops_redirect'])) {
+			$redirect = htmlspecialchars(trim($_GET['xoops_redirect']));
+			$isExternal = FALSE;
+			$tragetURL=substr(ICMS_URL,0,strpos(ICMS_URL,"/formulize")).$redirect;
+			redirect_header(ICMS_URL."/modules/formulize/application.php?id=all",5, _US_NOPERMISS."<br />Viewing: '".$tragetURL."'<br />");
```
Comment (eikonos) [inline user.php]: The error message would be better if it said, "Sorry, you do not have permission to view this page. Please contact your site administrator." Adding the link to the page is not necessary.

**Comment by jujinfu** [comment]
:+1:

**Pull request merged**